content/blog/model-calibration/index.Rmd

```diff
@@ -0,0 +1,409 @@
+
+Our plan is to implement model calibration in two phases: The first phase will focus
+on binary models, and the second phase will focus on multi-class models. 
+
+The first batch of enhancements are now available in the development version of 
+the probably package. The enhancements are centered around plotting functions meant for
+**diagnosing** the prediction's performance. These are more commonly known as 
+**calibration plots**.
+
+
+## Calibration Plots
+
+The idea behind a calibration plot is that if we group the predictions based on 
+their probability, then we should see an percentage of events ^[We can think of 
+an **event** as the outcome that is being tracked by the probability. For 
+example, in a model predicting "heads" or "tails", and we want to calibrate the
+probability for "tails", then the **event** is when the column containing the 
+outcome, has the value of "tails".]  that match such probability. 
+
+For example, if we collect a group of the predictions whose probabilities are 
+estimated to be about 10%. We should expect that about 10% of the those in the 
+group to indeed be events. The plots shown below can be used as diagnostics to 
+see if our predictions are consistent with the observed event rates. 
+
+
+### Example Data
+
+If you would like to follow along, load the probably and dplyr packages 
+into you R session.
+
+```{r}
+library(tidymodels)
+library(probably)
+```
+
+The probably package comes with a few data sets. For most of the examples in this post,
+we will use `segment_logistic`. It is an example data set that contains
+predictions, and their probabilities. `Class` contains the outcome of 
+"good" and "poor", `.pred_good` contains the probability that the event is "good". 
+
+```{r}
+segment_logistic
+```
+
+
+### Binned Plot
+
+On smaller data sets, it is a challenging to obtain an accurate *event rate* 
+for a given probability. For example, if there are 5 predictions with about a 50% 
+probability, and 3 of those are events, the plot would show a 60% event rate.
+This comparison would not be appropriate because there are not enough predictions
+to really determine how close to 50% the model really is. 
+
+The most common approach to group the probabilities
+into bins, or buckets.  Usually, the data is split into 10 discrete buckets,
+from 0 to 1 (0 - 100%). The *event rate* and the *bin midpoint* is 
+calculated for each bin. 
+
+In probably, binned calibration plots can be created using `cal_plot_breaks()`. 
+It expects a data set, and the un-quoted variable names that contains the events
+(`truth`), and the probabilities (`estimate`). For the example here, we pass the 
+`segment_logistic` data set, and use `Class` and `.pred_good` as the arguments. 
+By default, this function will create a calibration plot with 10 buckets (breaks):
+
+```{r, fig.alt = "Calibration plot with 10 bins, created with the cal_plot_breaks() function"}
```

Comment:
```suggestion
```{r, fig.alt = "A ggplot line plot with predicted probabilities on the x axis and event rates on the y axis, both ranging from 0 to 1. A dashed line lies on the identity line y equals x, and is loosely followed by a solid line that joins a series of dots representing the midpoint for each of 10 bins. Past predicted probabilities of 0.5, the dots consistently lie below the dashed line."}
```

Screen readers will be able to read the source code, so noting the function used in the chunk in the alt text will be read redundantly. You do a good job of narrating before/after the plot what the "message" ought to be in each, so that part is taken care of, but a bit more description of what's happening in each would be appreciated—a demo for this plot above!